packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,292 @@
+|MarketingSystem/MarketingVpc/PrivateSubnet1| Name | Platform|MarketingSystem/MarketingVpc/PrivateSubnet1|
+|MarketingSystem/MarketingVpc/PrivateSubnet1| aws-cdk:subnet-name | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet1| aws-cdk:subnet-type | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet1| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet1/RouteTable| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet1/RouteTable|Name| MarketingSystem/MarketingVpc/PrivateSubnet1|
+|MarketingSystem/MarketingVpc/PrivateSubnet2| Name | Platform|MarketingSystem/MarketingVpc/PrivateSubnet2|
+|MarketingSystem/MarketingVpc/PrivateSubnet2| aws-cdk:subnet-name | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet2| aws-cdk:subnet-type | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet2| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet2/RouteTable| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet2/RouteTable|Name| MarketingSystem/MarketingVpc/PrivateSubnet2|
+|MarketingSystem/MarketingVpc/PrivateSubnet3| Name | Platform|MarketingSystem/MarketingVpc/PrivateSubnet3|
+|MarketingSystem/MarketingVpc/PrivateSubnet3| aws-cdk:subnet-name | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet3| aws-cdk:subnet-type | Private|
+|MarketingSystem/MarketingVpc/PrivateSubnet3| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet3/RouteTable| CostCenter | Platform|
+|MarketingSystem/MarketingVpc/PrivateSubnet3/RouteTable|Name| MarketingSystem/MarketingVpc/PrivateSubnet3|
+|MarketingSystem/MarketingVpc/IGW|CostCenter|Platform|
+|MarketingSystem/MarketingVpc/IGW|Name|MarketingSystem/MarketingVpc|
+|MarketingSystem/B2BService/Service/SecurityGroup/Resource|CostCenter|Marketing|
+|MarketingSystem/B2BService/LB/Resource|CostCenter|Marketing|
+|MarketingSystem/B2BService/LB/SecurityGroup/Resource|CostCenter|Marketing|
+|MarketingSystem/B2BService/LB/PublicListener/ECSGroup/Resource|CostCenter|Marketing|
+|MarketingSystem/B2CService/Service/SecurityGroup/Resource|CostCenter|Marketing|
+|MarketingSystem/B2CService/LB/Resource|CostCenter|Marketing|
+|MarketingSystem/B2CService/LB/SecurityGroup/Resource|CostCenter|Marketing|
+|MarketingSystem/B2CService/LB/PublicListener/ECSGroup/Resource|CostCenter|Marketing|
+
+As you can see many tags are generated with only a few intent based directives. The CDK does default some additional tags for suggested `Name` keys. If you want to remove those tags you can do so by using the `RemoveTag` aspect, see below:
+
+```ts
+// snip //
+const vpc = new ec2.VpcNetwork(marketingStack, 'MarketingVpc', {
+  maxAZs: 3 // Default is all AZs in region
+  });
+// override the VPC tags with Platform
+// this will tag the VPC, Subnets, Route Tables, IGW, and NatGWs
+vpc.apply(new cdk.Tag(COST_CENTER_KEY, 'Platform'));
+vpc.apply(new cdk.RemoveTag('Name'));
+// snip //
+```
+
+If you run the above modification you will notice that the Subnets, Route
```

Comment:
You are correct in that the design I sent you towards has this issue. But rather than have a giant documentation section that explains the problem and gives you a horrible way to work around it, shouldn't we resolve the issue instead?

A simple solution would be to add an (optional) `priority` field to every Tag, and have it so that by default `RemoveTag.priority > Tag.priority` (so that `RemoveTag` takes precedence over `Tag` applied lower). We'll have to tombstone tags to make this work, but that doesn't seem too hard.